### PR TITLE
certificates unittest for cms and klass models

### PR DESCRIPTION
**Pull request by umarmughal824**
#### What are the relevant tickets?
#1115 

#### What's this PR do?
Certificates unittest for cms and klass models

### CODE REVIEW

File: klasses/models_test.py [hunk 1]
```diff
@@ -171,3 +193,29 @@ def test_total_due_by_next_deadline():
...
+    certificate = BootcampRunCertificateFactory.create(
+        user=user, bootcamp_run=bootcamp_run
+    )
+
+    assert certificate.is_revoked is False
+    assert certificate.start_end_dates == (
+        bootcamp_run.start_date,
+        bootcamp_run.end_date,
+    )
+    assert certificate.link == "/certificate/{}/".format(str(certificate.uuid))
```
Comment (HamzaIbnFarooq) [inline klasses/models_test.py:212]: nit: we should stick to one pattern for `format()` either use `{}` or `{uuid}`. secondly, we should also stay consistent with double `"` and single quotes `'`.

File: klasses/models_test.py [hunk 2]
```diff
@@ -171,3 +193,29 @@ def test_total_due_by_next_deadline():
...
+    assert str(
+        certificate
+    ) == 'BootcampRunCertificate for user={user}, run={bootcamp_run} ({uuid})"'.format(
```
Comment (HamzaIbnFarooq) [inline klasses/models_test.py:215]:
```suggestion
    ) == 'BootcampRunCertificate for user={user}, run={bootcamp_run} ({uuid})'.format(
```
Also remove that double-quotes from `BootcampRunCertificate.__str__` method

File: cms/models_test.py [hunk 3]
```diff
@@ -265,3 +270,64 @@ def test_home_catalog():
...
+    certificate = BootcampRunCertificateFactory.create(
+        bootcamp_run=bootcamp_run_page.bootcamp_run
+    )
+    certificate_page = CertificatePageFactory.create(parent=bootcamp_run_page)
+    request = rf.get(certificate_page.get_url())
+    assert (
+        certifcate_index_page.bootcamp_certificate(
+            request, certificate.uuid
+        ).status_code
+        == 200
```
Comment (HamzaIbnFarooq) [inline cms/models_test.py:326]: we can also use `status.HTTP_200_OK`

File: cms/factories.py [hunk 4]
```diff
@@ -289,3 +289,39 @@ class AdmissionSectionFactory(wagtail_factories.PageFactory):
...
+    slug = factory.fuzzy.FuzzyText()
+
+    class Meta:
+        model = models.CertificateIndexPage
+
+
+class CertificatePageFactory(wagtail_factories.PageFactory):
+    """CertificatePage factory class"""
+
+    bootcamp_run_name = factory.fuzzy.FuzzyText(prefix="bootcamp_run")
```
Comment (HamzaIbnFarooq) [inline cms/factories.py:319]:
```suggestion
    bootcamp_run_name = factory.fuzzy.FuzzyText(prefix="bootcamp_run_")
```

File: cms/models_test.py [hunk 5]
```diff
@@ -265,3 +270,64 @@ def test_home_catalog():
...
+    certificate_page = CertificatePageFactory.create(
+        parent=bootcamp_run_page,
+        bootcamp_run_name="bootcamp_run",
+        certificate_name="certificate_name",
+        location="location",
+        signatories__0__signatory__name="Name",
+        signatories__0__signatory__title_1="Title_1",
+        signatories__0__signatory__title_2="Title_2",
+        signatories__0__signatory__organization="Organization",
+        signatories__0__signatory__signature_image__title="Image",
```
Comment (HamzaIbnFarooq) [inline cms/models_test.py:292]: shouldn't we use SignatoryPageFactory here?

**Review by HamzaIbnFarooq — CHANGES_REQUESTED** [review]
commented some minor changes

File: cms/models_test.py (hunk 5 above)
Comment (umarmughal824) [reply cms/models_test.py:292]: See how it's being done in xPRO, you will get the idea of it.

File: cms/models_test.py (hunk 3 above)
Comment (umarmughal824) [reply cms/models_test.py:326]: We can use it but it's not necessary. As that status is a part of `rest_framework` but we are not using it anywhere in the `cms` application.

File: klasses/models_test.py (hunk 1 above)
Comment (umarmughal824) [reply klasses/models_test.py:212]: that `"` is also placed in the view so I kept it the same there. I did not notice that.
https://github.com/mitodl/bootcamp-ecommerce/blob/3d1df795f9a1360748e6f31b34e94c1723143b65/klasses/models.py#L343

File: klasses/models_test.py (hunk 2 above)
Comment (umarmughal824) [reply klasses/models_test.py:215]: that `"` is also placed in the view so I kept it the same there. I did not notice that. By the way, I have removed it in both 
https://github.com/mitodl/bootcamp-ecommerce/blob/3d1df795f9a1360748e6f31b34e94c1723143b65/klasses/models.py#L343

**Review by HamzaIbnFarooq — APPROVED** [review]
LGTM

**Pull request merged**